### PR TITLE
test(refs DPLAN-123): Add unit test for DpProgressBar

### PR DESCRIPTION
**Pull request by rafelddemos**
[DPLAN-123](https://demoseurope.youtrack.cloud/issue/DPLAN-123/Jede-demosplan-ui-Komponente-braucht-mindestens-einen-sinnvollen-Test-how-to-demosplan-ui-v1.0)

This PR adds a unit test for DpProgressBar component.

### CODE REVIEW

**Review by sakutademos — APPROVED** [review]
niiice, thank you so much for adding tests <3 just a small nitpick and a thought, maybe we could add some edge cases. I am not sure if it makes sense here, but what happens if an invalid value is passed?

File: tests/DpProgressBar.spec.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  const createWrapper = (props = {}) => {
+    return shallowMount(DpProgressBar, {
+      props: {
+        ...props,
+      },
+    })
+  }
+
+  const getProgressLine = () => {
```
Comment (hwiem) [inline tests/DpProgressBar.spec.js]: Nice that you extracted some helper functions ✨

**Review by hwiem — COMMENTED** [review]
Nice, thanks for adding this test! ❤️ Just 1 small concern

File: tests/DpProgressBar.spec.js (hunk 1 above)
Comment (rafelddemos) [reply tests/DpProgressBar.spec.js:15]: I saw it in a test you wrote :-)

File: tests/DpProgressBar.spec.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  afterEach(() => {
+    if (wrapper) {
+      wrapper.unmount()
+    }
+  })
+
+  describe('Rendering', () => {
+    it('renders label when provided', () => {
+      wrapper = createWrapper({ label: 'Loading...' })
```
[the remark this answers is not in the document]
Comment (rafelddemos) [reply tests/DpProgressBar.spec.js]: good idea, thanks! 
adjusted in 3a35e3a8df886b26aeabaf2e22e44f41470aea3f

**Comment by hwiem** [comment]
Now that we migrated from jest to vitest, this test probably needs a few small changes before merging :disappointed: 

It seems I got lucky and only the import had to be adjusted :) changed in 3e1f744330a63d466c9b4f4359d97dec43b7a86a

File: tests/DpProgressBar.spec.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    })
+  }
+
+  const getProgressLine = () => {
+    return wrapper.find('.c-progress-bar__line')
+  }
+
+  const expectProgressWidth = (expectedWidth) => {
+    const progressLine = getProgressLine()
+    expect(progressLine.attributes('style')).toContain(`width: ${expectedWidth}%`)
```
[the remark this answers is not in the document]
Comment (rafelddemos) [reply tests/DpProgressBar.spec.js]: good idea, thanks! changed in 3e1f744330a63d466c9b4f4359d97dec43b7a86a

**Comment by rafelddemos** [comment]
> niiice, thank you so much for adding tests <3 just a small nitpick and a thought, maybe we could add some edge cases. I am not sure if it makes sense here, but what happens if an invalid value is passed?

Hmm, I guess it makes sense to add tests for `NaN` and `null` (since it returns an empty String and the default `0`). Added in 3e1f744330a63d466c9b4f4359d97dec43b7a86a

File: tests/DpProgressBar.spec.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import DpProgressBar from '~/components/DpProgressBar/DpProgressBar.vue'
```
Comment (hwiem) [inline tests/DpProgressBar.spec.js]:
```suggestion
import DpProgressBar from '~/components/DpProgressBar/DpProgressBar'
```

**Review by hwiem — APPROVED** [review]
Just a tiny nitpick, otherwise looks good :)
